src/lavinmq/queue/stream_queue_message_store.cr

```diff
@@ -88,6 +106,83 @@ module LavinMQ
         {msg_offset, segment, pos}
       end
 
+      def last_offset_by_consumer_tag(consumer_tag : String)
+        if pos = @consumer_offset_positions[consumer_tag]?
+          tx = @consumer_offsets.to_slice(pos, 8)
+          return IO::ByteFormat::LittleEndian.decode(Int64, tx)
+        end
+      end
+
+      private def restore_consumer_offset_positions : Hash(String, Int64)
+        positions = Hash(String, Int64).new
+        return positions if @consumer_offsets.size.zero?
+
+        loop do
+          ctag = AMQ::Protocol::ShortString.from_io(@consumer_offsets)
+          break if ctag.empty?
+          positions[ctag] = @consumer_offsets.pos
+          @consumer_offsets.skip(8)
+        rescue IO::EOFError
+          break
+        end
+        @consumer_offsets.pos = 0 if @consumer_offsets.pos == 1
+        @consumer_offsets.resize(@consumer_offsets.pos)
```

Comment:
so then there's no room left to expand in?

Reply:
Write_bytes writes at the end of the file, and doesn't care about pos, so we need to resize it to pos.